src/renderer/views/stake/StakeView.tsx

```diff
@@ -12,28 +14,44 @@ import { useParams } from 'react-router-dom'
 import { AddStakeStory } from '../../components/stake/AddStake/AddStake.stories'
 import { Stake } from '../../components/stake/Stake'
 import BackLink from '../../components/uielements/backLink'
-/**
- * temporary data mock
- * @TODO @thatStrangeGuyThorchain replace mocck after
- * https://github.com/thorchain/asgardex-electron/issues/441 resolved
- */
-import { defaultPoolShare } from '../../components/uielements/poolShare/PoolShare.stories'
+import { useBinanceContext } from '../../contexts/BinanceContext'
 import { useMidgardContext } from '../../contexts/MidgardContext'
+import { useWalletContext } from '../../contexts/WalletContext'
 import { StakeRouteParams } from '../../routes/stake'
 import { PoolDetailsView } from './PoolDetails/PoolDetailsView'
+import { ShareView } from './Share/ShareView'
 
 type Props = {}
 
 const StakeView: React.FC<Props> = (_) => {
   const { asset: assetParam } = useParams<StakeRouteParams>()
   const { service: midgardService } = useMidgardContext()
+  const { keystoreService } = useWalletContext()
+  const { address$ } = useBinanceContext()
 
-  const asset = assetFromString(assetParam)
+  const keystore = useObservableState(keystoreService.keystore$, O.none)
 
-  useEffect(() => {
-    midgardService.pools.reloadPoolDetailedState(O.fromNullable(asset))
-  }, [asset, midgardService.pools])
+  const hasWallet = useMemo(
+    () =>
+      pipe(
+        keystore,
+        O.flatten,
+        O.map(Boolean),
+        O.getOrElse(() => false)
+      ),
+    [keystore]
+  )
```

Comment:
^ Sounds like a same state we can get with [!hasImportedKeystore](https://github.com/thorchain/asgardex-electron/blob/4686fd192bb2efc692a5ef01034a347b138ecbb9/src/renderer/services/wallet/util.ts#L29), isn't?

Reply:
not exactly. this is like [isLocked](https://github.com/thorchain/asgardex-electron/blob/4686fd192bb2efc692a5ef01034a347b138ecbb9/src/renderer/services/wallet/util.ts#L31) value :)

Reply:
I'm still not sure if `isLocked` is correct - see https://github.com/thorchain/asgardex-electron/pull/488#discussion_r494547861